File: armstrong-number/armstrong.c
#include <stdio.h>
#include <math.h>

int numOfDigit(int num) {
   int count = 0;

   while(num != 0) {
     num /= 10;
     count++;
   }
   return count;
}

int main() {

  int number, originalNumber, remainder, result = 0;

  printf("Enter a number: ");
  scanf("%d", &number);

  int digitCount = numOfDigit(number);

  originalNumber = number;

  while(originalNumber != 0) {
    remainder = originalNumber % 10;
    result += pow(remainder, digitCount);
    originalNumber /= 10;
  }
    if (number == result) {
    printf("%d is armstrong number\n", number);
  } else {
    printf("%d is not armstrong number\n", number);
  }
}

File: armstrong-number/armstrong.js
function numOfDigit(num) {
  let count = 0;
  while (num !== 0) {
    num = parseInt(num / 10);
    count++;
  }

  return count;
}

let number,
  originalNumber,
  remainder,
  result = 0;

console.log("Enter a number");
number = parseInt(process.argv[2]);
const digitCount = numOfDigit(number);

originalNumber = number;

while (originalNumber !== 0) {
  remainder = parseInt(originalNumber % 10);
  result += Math.pow(remainder, digitCount);
  originalNumber =  parseInt(originalNumber/10);
}

if (result === number) {
  console.log(`${number} is armstrong`);
} else {
  console.log(`${number} is not armstrong`);
}
